database-plugins/src/main/java/io/cdap/plugin/db/common/DBDifferenceUtils.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.common;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import javax.annotation.Nullable;
+
+/**
+ * A DB Util that handles the difference of different DB
+ */
+public final class DBDifferenceUtils {
+  public static final String DB_PRODUCT_NAME_POSTGRESQL = "POSTGRESQL";
+  public static final String DB_PRODUCT_NAME_ORACLE = "ORACLE";
+  public static final String DB_PRODUCT_NAME_SQLSERVER = "MICROSOFT SQL SERVER";
+  public static final String RESULTSET_COLUMN_TABLE_CAT = "TABLE_CAT";
+  private static final String SQLSERVER_DATABSE_PARAM = "databaseName=";
+
+  private DBDifferenceUtils() {
+  }
+
+  /**
+   * Replace the database name in the JDBC connection string with specified database name
+   *
+   * @param connectionString the JDBC connection string
+   * @param database         the database name
+   * @return the replaced JDBC connection string
+   */
+  public static String replaceDatabase(String connectionString, @Nullable String database) {
```

Comment:
As discussed offline, let's avoid this database replacing and make the user pass in a correct connection string.

Reply:
done :)